Shooter enemies that actually hurt the player

3akev opened the pull request:
- Also a sane collision layer/mask setup

nightblade9 commented:
I suspect the `main.gd` changes and the jump fix is because we're using different versions of Godot. I'm using `v3.0.2.stable.official` according to `Help > About`, how about you?

ChemicalInk commented:
I'm using the same version. LGTM, and sorry for the delay... Guess I didn't jump enough? :/

nightblade9 committed:
Created shooter enemy that shoots bullets.
Bullets just shoot in the wrong direction.

nightblade9 committed:
Fix enemy bullet angle
Enemy bullets are now a separate type
TODO: bullets shouldn't collide with shooter; should collide with player

nightblade9 committed:
Parameterized collider group for bullets.
Added "enemybullets" group too.

nightblade9 committed:
Fix bullet groups by splitting out base vs. player vs. enemy

Commit:
Overhauled collision layers/masks

Commit:
Bullets properly damage things again

Commit:
Merge with master

nightblade9 commented on prototype/Bullet/PlayerBullet.gd:
```diff
@@ -1,19 +1,5 @@
-# var b = "textvar"
-
 func init(x, y, angle):
 	.init("enemies", x, y, angle)
```
We can probably remove this file.
